packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -165,3 +167,28 @@ export function DatePicker({
     setOpen(false);
   }
 }
+
+function useEscapeKeyToCloseDatePicker(
+  open: boolean,
+  ref: React.RefObject<HTMLDivElement>,
+): { pickerRef: React.RefObject<ReactDatePicker> } {
+  const pickerRef = useRef<ReactDatePicker>(null);
+
+  const escFunction = (event: KeyboardEvent) => {
+    if (event.key === "Escape" && open) {
+      pickerRef.current?.setOpen(false);
+      event.stopPropagation();
```

Comment:
should probably explain this since it has a big role in this whole thang

Reply:
shore!